gsk4/src/render_node.rs

```diff
@@ -0,0 +1,171 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::RenderNodeType;
+use glib::translate::*;
+use glib::StaticType;
+use std::fmt;
+use std::ptr;
+
+glib::wrapper! {
+    #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+    pub struct RenderNode(Shared<ffi::GskRenderNode>);
+
+    match fn {
+        ref => |ptr| ffi::gsk_render_node_ref(ptr),
+        unref => |ptr| ffi::gsk_render_node_unref(ptr),
+    }
+}
+
+impl StaticType for RenderNode {
+    fn static_type() -> glib::Type {
+        unsafe { from_glib(ffi::gsk_render_node_type_get_type()) }
+    }
+}
+
+impl RenderNode {
+    //#[doc(alias = "gsk_render_node_deserialize")]
+    //pub fn deserialize(bytes: &glib::Bytes, error_func: /*Unimplemented*/FnMut(/*Ignored*/ParseLocation, /*Ignored*/ParseLocation, &glib::Error), user_data: /*Unimplemented*/Option<Fundamental: Pointer>) -> Option<RenderNode> {
+    //    unsafe { TODO: call ffi:gsk_render_node_deserialize() }
+    //}
+
+    pub fn downcast<T: IsARenderNode>(self) -> Result<T, RenderNode> {
+        unsafe {
+            if self.get_type() == T::static_type() {
+                Ok(from_glib_full(self.to_glib_full()))
+            } else {
+                Err(self)
+            }
+        }
+    }
+
+    pub fn downcast_ref<T: IsARenderNode>(&self) -> Option<&T> {
+        unsafe {
+            if self.get_type() == T::static_type() {
+                Some(&*(self as *const RenderNode as *const T))
+            } else {
+                None
+            }
+        }
+    }
+
+    pub fn get_type(&self) -> glib::Type {
+        Self::static_type()
+    }
```

Comment:
This doesn't make much sense, as you should be using something else for downcast/downcast_ref

Reply:
Yup, I got rid of that method.